libzfs.pyx

```diff
@@ -2595,18 +2595,25 @@ cdef class ZFSPool(object):
                 ELSE:
                     return PoolStatus(libzfs.zpool_get_status(self.handle, &msg_id))
 
+    def __warning_statuses(self):
+        statuses = [
+            PoolStatus.RESILVERING,
+            PoolStatus.VERSION_OLDER,
+            PoolStatus.FEAT_DISABLED,
+        ]
+
+        IF HAVE_ZPOOL_STATUS_NON_NATIVE_ASHIFT:
+            statuses.append(PoolStatus.NON_NATIVE_ASHIFT)
+
+        return statuses
+
     property healthy:
         def __get__(self):
-            ok = [
-                PoolStatus.OK,
-                PoolStatus.VERSION_OLDER,
-                PoolStatus.FEAT_DISABLED
-            ]
+            return self.status_code in [PoolStatus.OK] + self.__warning_statuses()
 
-            IF HAVE_ZPOOL_STATUS_NON_NATIVE_ASHIFT:
-                ok.append(PoolStatus.NON_NATIVE_ASHIFT)
-
-            return self.status_code in ok
+    property warning:
```

Comment:
We probably want to add this to `__getstate__` like we have for `healthy` property ?

Reply:
Actually we should definitely do the above as middleware PR should fail otherwise ?